### ISSUE TITLE
Error recommendation doesn't work for enum values

### ISSUE THREAD

**Issue by jiasli** [Azure/azure-cli#17395]
**Describe the bug**

Error recommendation doesn't work for enum values. 

**To Reproduce**

For example, when I provide `storage` for `--resource-type`:

```
> az account get-access-token --resource-type storage
az account get-access-token: 'storage' is not a valid value for '--resource-type'.

TRY THIS:
az account get-access-token
Get an access token for the current account

https://docs.microsoft.com/en-US/cli/azure/account#az_account_get_access_token
Read more about the command in reference docs
```

Instead of telling me the allowed values, the recommendation simply removes `--resource-type` which is not helpful.

**Expected behavior**

```
> az account get-access-token --resource-type storage
az account get-access-token: 'storage' is not a valid value for '--resource-type'.
Allowed values: aad-graph, arm, batch, data-lake, media, ms-graph, oss-rdbms.

TRY THIS:
az account get-access-token --resource-type arm
```

As there is a chance that the content of `TRY THIS:` is not helpful, so `TRY THIS:` should be changed:

```diff
- TRY THIS:
+ Examples from AI knowledge base:
```

**Additional context**

The more predicable approach is proposed by https://github.com/microsoft/knack/pull/237.

**Comment by RakeshMohanMSFT** [comment]
This is fixed.
az account get-access-token --resource-type storage
az account get-access-token: 'storage' is not a valid value for '--resource-type'. Allowed values: oss-rdbms, arm, aad-graph, ms-graph, batch, media, data-lake.

**Comment by jiasli** [comment]
Fixed by #17621
